Module.lua

```diff
@@ -348,3 +348,25 @@ function Module:listModules()
    end
    return modules
 end
+
+function Module:clearState()
+    if self.output then
+       if torch.isTensor(self.output) then
+          self.output = self.output.new()
```

Comment:
Rather than creating a new tensor, isn't it better to do `self.output:set()` which resets the tensor? Then all references to it are kept intact.